hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -403,7 +403,7 @@ private static Boolean tableExists(Connection conn, Map<String, String> options)
       statement.setQueryTimeout(Integer.parseInt(options.get(JDBCOptions.JDBC_QUERY_TIMEOUT())));
       statement.executeQuery();
     } catch (SQLException e) {
-      return false;
+      e.printStackTrace();
```

Comment:
+1 for printing the original exception information. A better way is to throw the exception but should not let it return true?

Reply:
I have considered throwing exceptions instead of printing exception information directly, but I haven't figured out the benefits of throwing exceptions.In other words, how to better handle exception information after throwing an exception

Reply:
First of all, re-throw an exception means here we may not process it correctly (e.g. you can not judge that if the table exists.). And we can let the outer process it or exits naturally if necessary (in the mode, we can also see the exception detail). 

For printing exception details, a better way is to use the logging framework, if you want to print.

Reply:
Is this OK now? The exception information will be printed like this
```java
xception in thread "main" org.apache.hudi.exception.HoodieException: Failed to get Schema through jdbc. 
	at org.apache.hudi.utilities.schema.JdbcbasedSchemaProvider.getSourceSchema(JdbcbasedSchemaProvider.java:81)
	at org.apache.hudi.utilities.schema.SchemaProviderWithPostProcessor.getSourceSchema(SchemaProviderWithPostProcessor.java:42)
	at org.apache.hudi.utilities.deltastreamer.DeltaSync.registerAvroSchemas(DeltaSync.java:865)
	at org.apache.hudi.utilities.deltastreamer.DeltaSync.<init>(DeltaSync.java:235)
	at org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer$DeltaSyncService.<init>(HoodieDeltaStreamer.java:658)
	at org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer.<init>(HoodieDeltaStreamer.java:143)
	at org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer.<init>(HoodieDeltaStreamer.java:116)
	at org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer.main(HoodieDeltaStreamer.java:557)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:845)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:161)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:184)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:920)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:929)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: org.apache.hudi.exception.HoodieException: org.apache.hive.service.cli.HiveSQLException: Error while compiling statement: FAILED: HiveAccessControlException Permission denied: user [spark] does not have [SELECT] privilege on [test/test_source_json/*]
	at org.apache.hudi.utilities.UtilHelpers.tableExists(UtilHelpers.java:405)
	at org.apache.hudi.utilities.UtilHelpers.getJDBCSchema(UtilHelpers.java:421)
	at org.apache.hudi.utilities.schema.JdbcbasedSchemaProvider.getSourceSchema(JdbcbasedSchemaProvider.java:79)
	... 19 more
Caused by: org.apache.hive.service.cli.HiveSQLException: Error while compiling statement: FAILED: HiveAccessControlException Permission denied: user [spark] does not have [SELECT] privilege on [test/test_source_json/*]
	at org.apache.hive.jdbc.Utils.verifySuccess(Utils.java:255)
	at org.apache.hive.jdbc.Utils.verifySuccessWithInfo(Utils.java:241)
	at org.apache.hive.jdbc.HiveStatement.execute(HiveStatement.java:254)
	at org.apache.hive.jdbc.HiveStatement.executeQuery(HiveStatement.java:392)
	at org.apache.hive.jdbc.HivePreparedStatement.executeQuery(HivePreparedStatement.java:109)
	at org.apache.hudi.utilities.UtilHelpers.tableExists(UtilHelpers.java:403)
	... 21 more
Caused by: org.apache.hive.service.cli.HiveSQLException: Error while compiling statement: FAILED: HiveAccessControlException Permission denied: user [spark] does not have [SELECT] privilege on [test/test_source_json/*
...
```

Reply:
LGTM, after the CI pass, ping me again~

Reply:
> CI
@yanghua Hello,the CI has all passed